viewing_party/party.py

```diff
@@ -1,23 +1,219 @@
+    :params: dict - user_data: "watched" movies
+             dict - movie: movie details
+    :returns: nested dict - updated_data: updated watched movies 
+    """
+    # copy user_data
+    updated_data = user_data.copy()
+    # update watched movie
+    updated_data["watched"] = [movie]
+    return updated_data
+
+
+def add_to_watchlist(user_data, movie):
+    """
+    Updates dataset to include movies to be watched and details
+    :params: dict - user_data: "watchlist" movies
+             dict - movie: movie details
+    :returns: nested dict - updated_data: updated watchlist movies
+    """
+    # copy user data
+    updated_data = user_data.copy()
+    # update watchlist movie
+    updated_data["watchlist"] = [movie]
+    return updated_data
+
+def watch_movie(janes_data, MOVIE_TITLE_1):
+    """
+    Moves a movie from watchlist to watched
+    :params: dict - janes_data: dataset containing watched and watchlist movies
+             str - MOVIE_TITLE_1: movie title
+    :returns: nested dict - updated_data: updated watched/watchlist movies
+    """
+    # copy user data
+    updated_data = janes_data.copy()
+    # iterate through watchlist movies
+    for movie in updated_data['watchlist']:
+        # assume movie title == MOVIE_TITLE_1
+        if movie['title'] == MOVIE_TITLE_1:
+            # remove from watchlist
+            updated_data['watchlist'].remove(movie)
+            # add to watched
+            updated_data['watched'].append(movie)
+    return updated_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(janes_data):
+    """
+    Calculates movie rating average in watched movies
+    :params: dict - janes_data: dataset containing watched and watchlist movies
+    :returns: float - average: movie rating average
+    """
+    # collect ratings of movies watched
+    ratings = []
+    for movie in janes_data['watched']:
+        ratings.append(movie['rating'])
+    if len(ratings) == 0:
+        return 0
+    # calculate the average
+    average = sum(ratings)/len(ratings)
+    return average
+
+    
+def get_most_watched_genre(janes_data):
+    """
+    Collects watched movie genres and generates most popular genre by calculating the mode.
+    :params: dict - janes_data: dataset containing watched and watchlist movies
+    :returns: str - popular_genre: most popular movie genre watched
+    """
+    # collect genres of movies watched
+    genres = []
+    for movie in janes_data['watched']:
+        genres.append(movie['genre'])
+    if len(genres) == 0:
+        return None
+    # collect highest frequency movie genre
+    popular_genre = statistics.mode(genres)
+    return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(amandas_data):

```

Comment:
Good solution! Can you think of how sets might be used for an even simpler implementation? Though, I do think it's useful to have gone through the foundations and understanding what might be going on behind the scenes with sets.